### PR TITLE
LibGfx/JPEG: Refill reservoir if necessary when discarding bits

### PR DESCRIPTION
**Pull request by tcl3**
Running the JPEG decoder on [this corpus of 109,233 images](https://digitalcorpora.s3.amazonaws.com/corpora/files/govdocs1/by_type/jpg.zip), found 157 images that hit this `TODO()`. This change makes all of these images load correctly.

This was also the only thing that caused the JPEG decoder to crash when attempting to decode these files.

Fixes oss fuzz issue [63815](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63815)

### CODE REVIEW

**Comment by LucasChollet** [comment]
I'm away from home, and I don't want to download 35 Go on my mobile data. To clarify, are all of these images (especially the 157 crashing) valid or does the corpus include fail cases?


No doubt that I wasn't able to find one manually when you see the proportion on a quite significant set of samples 😅.

**Comment by tcl3** [comment]
> I'm away from home, and I don't want to download 35 Go on my mobile data. To clarify, are all of these images (especially the 157 crashing) valid or does the corpus include fail cases?

The corpus doesn't deliberately include images that fail to decode. All 157 of the images that hit this condition are valid and decode successfully for me on Linux.

Prior to this change, of the 109,233 total images:
* 108,182 decoded successfully
* 47 weren't recognized as JPEGs by `Gfx::ImageDecoder::try_create_for_raw_bytes()`.
   * 3 of these failed to decode on Linux for me.
* 1,004 failed to decode (the call to `decoder->frame(0)` failed.)
   * 2 of these failed to decode on Linux for me.

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp [hunk 1]
```diff
@@ -296,11 +296,8 @@ class HuffmanStream {
     {
         m_bit_offset += count;
 
-        if (m_bit_offset > bits_in_reservoir) {
-            // FIXME: I can't find a test case for that so let's leave it for later
-            //        instead of inserting an hard-to-find bug.
-            TODO();
-        }
+        if (m_bit_offset > bits_in_reservoir)
+            TRY(refill_reservoir());
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:300]: I initially wrote something for this case but discarded it when I discovered that this was not covered by my test cases. And I'm sure it was longer than that, but after a day I think that I finally remember why, I had a unique logic to handle the case where `m_bit_offset` was bigger than the reservoir itself. But `refill_reservoir()` seems to handle that just fine. So, I guess than one of my early version didn't.

**Review by LucasChollet — APPROVED** [review]
It's so simple that it's almost suspicious!

**Pull request merged**